wren-ui/src/pages/api/ask_task/streaming_answer.ts

```diff
@@ -0,0 +1,132 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { components } from '@/common';
+import { ThreadResponseAnswerStatus } from '@/apollo/server/services/askingService';
+
+const { wrenAIAdaptor, askingService } = components;
+
+class ContentMap {
+  private contentMap: { [key: string]: string } = {};
+
+  // Method to append (concatenate) content to the map
+  public appendContent(key: string, content: string) {
+    if (!this.contentMap[key]) {
+      this.contentMap[key] = '';
+    }
+    this.contentMap[key] += content;
+  }
+
+  // Method to get content from the map
+  public getContent(key: string): string | undefined {
+    return this.contentMap[key];
+  }
+
+  // Method to remove content from the map
+  public remove(key: string) {
+    delete this.contentMap[key];
+  }
+}
+
+const contentMap = new ContentMap();
+
+export default async function handler(
```

Comment:
should we restrict the request method?

Reply:
resolved in [2b34fee](https://github.com/Canner/WrenAI/pull/986/commits/2b34fee36fc94df587dceb8e93ee6ce53b846efa)